lib/mysql.js

```diff
@@ -331,12 +331,26 @@ MySQL.prototype.getInsertedId = function(model, info) {
  * @returns {*} The escaped value of DB column
  */
 MySQL.prototype.toColumnValue = function(prop, val) {
-  if (val == null) {
-    if (prop.autoIncrement || prop.id) {
-      return new ParameterizedSQL('DEFAULT');
-    }
+  if (val === undefined && this.isNullable(prop)) {
     return null;
   }
+  if (val === null) {
+    if (this.isNullable(prop)) {
+      return val;
+    } else {
+      try {
+        var castNull = prop.type(val);
+        if (prop.type === Object) {
+          return JSON.stringify(castNull);
+        }
+        return castNull;
+      } catch (err) {
+        //if we can't coerce null to a certain type,
+        //we just return it
+        return 'null';
```

Comment:
If this is specifically to make a query string where the value is null, and it fails our `val === null` and `.isNullable` checks, why not always set the string to null instead of attempting to stringify something?